include/smack/AnnotateLoopExits.h

```diff
@@ -0,0 +1,24 @@
+//
+// This file is distributed under the MIT License. See LICENSE for details.
+//
+#ifndef ANNOTATELOOPEXITS_H
+#define ANNOTATELOOPEXITS_H
+
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Pass.h"
+#include <map>
+
+namespace smack {
+
+class AnnotateLoopExits : public llvm::ModulePass {
```

Comment:
Why should this be a module pass, as opposed to a more fine-grained pass, e.g., function or loop pass: https://llvm.org/docs/WritingAnLLVMPass.html#the-looppass-class

Generally speaking, you want to reduce the scope as much as possible — see docs above.